test/TestIntegration.java

```diff
@@ -1004,6 +965,181 @@ public void prefetchMeta() throws Exception {
     assertNotNull(region_info);
   }
 
+
+  /** Basic reverse scan test. Modeled after the basic forward scan. */
+  @Test
+  public void basicReverseScan() throws Exception {
+    client.setFlushInterval(FAST_FLUSH);
+    final PutRequest put1 = new PutRequest(table, "brs1", family, "q", "v1");
+    final PutRequest put2 = new PutRequest(table, "brs2", family, "q", "v2");
+    final PutRequest put3 = new PutRequest(table, "brs3", family, "q", "v3");
+    Deferred.group(client.put(put1), client.put(put2),
+            client.put(put3)).join();
+    // Scan the same 3 rows created above twice.
+    for (int i = 0; i < 2; i++) {
```

Comment:
Yes this function performs the same scan twice. The forward scan test that they had written does the same thing. I think this is to ensure that scanners can be opened on the same rows without error and that the call to nextRows(1) is working in that only one row gets scanned at a time.